src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+    {
+        Unspecified,
+        Sequential
+    }
+
+    public static class CollectionAssert
+    {
+        public static void Equal(IEnumerable expected, IEnumerable actual)
+        {
+            Assert.Equal(expected.OfType<object>(), actual.OfType<object>());
+        }
+
+        public static void Equal(IList expected, IList actual)
+        {
+            Equal((IEnumerable)expected, actual);
+            Assert.Equal(expected.Count, actual.Count);
+            for (int i = 0; i < expected.Count; i++)
+            {
+                var expectedArray = expected as Array;
+                var actualArray = expected as Array;
+                var expectedLowerBound = expectedArray == null
+                                             ? 0
+                                             : expectedArray
+                                                   .GetLowerBound(0);
+                var actualLowerBound = actualArray == null
+                                           ? 0 : actualArray.GetLowerBound(
+                                               0);
+                Assert.Equal(expected[i + expectedLowerBound], actual[i + actualLowerBound]);
+            }
+        }
+    }
+
+    public static class IListExtensions
+    {
+        public static void AddRange(this IList list, IEnumerable items)
+        {
+            foreach (var item in items)
+                list.Add(item);
+        }
+
+        public static void InsertRange(this IList list, object[] items)
+        {
+            for (int i = 0; i < items.Length; i++)
+                list.Insert(i, items[i]);
+        }
+
+        public static void InsertRange(this IList list, int listStartIndex, object[] items, int startIndex, int count)
+        {
+            var numToInsert = items.Length - startIndex;
+            if (count < numToInsert)
+                numToInsert = count;
+            for (int i = 0; i < numToInsert; i++)
+                list.Insert(listStartIndex + i, items[startIndex + i]);
+        }
+
+        public static void AddRange(this IList list, object[] items, int startIndex, int count)
+        {
+            var numToAdd = items.Length - startIndex;
+            if (count < numToAdd)
+                numToAdd = count;
+            for (int i = 0; i < numToAdd; i++)
+                list.Add(items[startIndex + i]);
+        }
+    }
+
+    public abstract class IListTest<T> : ICollectionTest<T>
+    {
+        protected bool ExpectedIsReadOnly { get; private set; }
+        protected bool ExpectedIsFixedSize { get; private set; }
+        protected bool SearchingThrowsFromInvalidValue { get; set; }
+        protected IListTest(bool isSynchronized, bool isReadOnly, bool isFixedSize) : base(isSynchronized)
+        {
+            ExpectedIsReadOnly = isReadOnly;
+            ExpectedIsFixedSize = isFixedSize;
+            SearchingThrowsFromInvalidValue = false; // This SHOULD be false but this is the current behavior
```

Comment:
That comment is left over from the old tests, It looks like we fixed the behavior. I'll remove it.